### ISSUE TITLE
positive/negative ratio when there are fewer ground truth for each image. 

### ISSUE THREAD

**Issue by qiuyuchen14** [facebookresearch/maskrcnn-benchmark#1251]
## ❓ Questions and Help
Hi, I'm trying to train the model on my dataset that only has one or two ground-truth objects in each image. The idea is to only predict objects conditioned in a certain region. I found my positive # is around 5-10, but negative # is ~500, which gives me a huge imbalance issue. I tried to fix the positive/negative ratio to 1:3, and increase the batch size, but it didn't seem to help: during training, the prediction looks reasonable, but during test time (even test on the same training set), the prediction looks terrible (The training loss is going down). I think there's still some issue with my positive/negative ratio. I have a couple of questions here: 

1. I found in rpn, it compares IoU of anchors with GT, and take IoU>0.7 as positives and IoU<0.3 as negatives, and throw away anchors with IoU between 0.3 and 0.7. Why throwing them away? I tested it if I only sample negatives from 0.3-0.7, it always gives me many small boxes on the boundary of the image. I'm curious why. 
2. For my application, since I want to make a prediction based on a condition of an image coordinate, and I expect just one or two objects in this region. I was hoping to sample more negatives around this region, fewer negatives in the irrelevant region of the image. Does it make sense to sample negatives from 0.3-0.7, instead of throwing them away? I could tweak the threshold value a bit. 

Thanks!!
